iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2606,6 +2606,12 @@ struct  InstanceXmlReader
             if (ECObjectsStatus::Success != SchemaKey::ParseSchemaFullName(key, m_fullSchemaName.c_str()))
                 return NULL;
 
+            if (m_context.IsSchemaToBePruned(key))

```

Comment:
I don't think you have changed the behavior at all.  This method will still return null and then GetInstance will log an error because GetSchema returned a nullptr.